app/models/invoice.rb

```diff
@@ -13,6 +13,17 @@ def self.unshipped_items
     .order('invoices.created_at asc')
   end
 
+  def self.highest_revenue_date(item_id)
+    joins(:invoice_items)
+    .joins('right join transactions on transactions.invoice_id=invoices.id')
+    .select(invoices: :updated_at)
+    .where(invoice_items: {item_id: item_id}, transactions: {result: 1}, invoices: {status: 1})
+    .order(Arel.sql('sum(invoice_items.quantity * invoice_items.unit_price) desc, invoices.updated_at desc'))
```

Comment:
@mkrumholz potential refactor to calculate the revenue in the `.select `with an `AS revenue` & then call it in the `.order` without needing the `AREL.sql`

Reply:
@Sidarth20 I just went and tried this, but it causes an error. Honestly not sure why, but it won't let me use the alias in my order clause.